test/e2e/suite_log_test.go

```diff
@@ -0,0 +1,99 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util/names"
+	"github.com/karmada-io/karmada/pkg/util/restmapper"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+)
+
+const logDir = "/tmp/karmada/objects"
+
+func printAllBindingAndRelatedObjects() {
+	filePath := filepath.Join(logDir, testNamespace+".txt")
+	err := os.MkdirAll(logDir, 0700)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	logFile, err := os.OpenFile(filePath, os.O_CREATE|os.O_APPEND|os.O_RDWR, 0600)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	defer logFile.Close()
+
+	rbList, err := karmadaClient.WorkV1alpha2().ResourceBindings(testNamespace).List(context.TODO(), metav1.ListOptions{})
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	for _, rb := range rbList.Items {
+		// print resource binding to log file
+		printObjectToFile(logFile, &rb)
+		printBindingRelatedResourceAndWork(logFile, rb.ObjectMeta, rb.Spec, workv1alpha2.ResourceBindingPermanentIDLabel)
+	}
+
+	crbList, err := karmadaClient.WorkV1alpha2().ClusterResourceBindings().List(context.TODO(), metav1.ListOptions{})
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	for _, crb := range crbList.Items {
+		// print cluster resource binding to log file
+		printObjectToFile(logFile, &crb)
+		printBindingRelatedResourceAndWork(logFile, crb.ObjectMeta, crb.Spec, workv1alpha2.ClusterResourceBindingPermanentIDLabel)
+	}
+}
+
+func printBindingRelatedResourceAndWork(file *os.File, metadata metav1.ObjectMeta, spec workv1alpha2.ResourceBindingSpec, permanentIDLabelKey string) {
+	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(spec.Resource.APIVersion, spec.Resource.Kind))
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	resource, err := dynamicClient.Resource(gvr).Namespace(spec.Resource.Namespace).Get(context.TODO(), spec.Resource.Name, metav1.GetOptions{})
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	// print resource template to log file
+	printObjectToFile(file, resource)
+
+	for _, clusterName := range framework.ClusterNames() {
+		workNamespace := names.GenerateExecutionSpaceName(clusterName)
+		workList, err := karmadaClient.WorkV1alpha1().Works(workNamespace).List(context.TODO(), metav1.ListOptions{
```

Comment:
If namespaces are not distinguished, can a list request query all target work objects?

Reply:
it‘s surprising, refer to https://github.com/kubernetes/client-go/issues/159#issuecomment-288624475, it really can query all objects in just one list request.